agents/dev/critical_processes_check.py

```diff
@@ -0,0 +1,168 @@
+"""
+Universal critical-process monitor for any ServerAgent.
+
+Usage:
+    python critical_processes_check.py <agent_type> [--interval N]
+
+Example:
+    python critical_processes_check.py ntp --interval 5
+"""
+
+import argparse
+import logging
+import os
+import sys
+import threading
+import time
+
+from agents.agent import ServerAgent  # for type‐checking only
+
+
+def import_agent_class(agent_type):
+    """
+    Dynamically import and return the Agent class for a given type.
+
+    Args:
+        agent_type (str): Lowercase name of the agent module,
+        e.g. 'ntp', 'dns', 'smtp'.
+
+    Returns:
+        type: The Agent subclass (e.g. NTPAgent, DNSAgent).
+
+    Raises:
+        ImportError: If the module cannot be imported.
+        AttributeError: If the expected class is not found in the module.
+    """
+    module_name = 'agents.{0}.{0}'.format(agent_type)
+    class_name = '{0}Agent'.format(agent_type.upper())
+    module = __import__(module_name, fromlist=[class_name])
+    return getattr(module, class_name)
+
+
+def monitor_processes(agent, interval, stop_event):
```

Comment:
`is_service_healthy` is to wrap a dedicated method, which checks all critical processes. 
`monitor_processes` worker should do either one or the other: check health status or check processes.